sway-lib-std/src/asset_id.sw

```diff
@@ -1,17 +1,116 @@
+    ///
+    /// # Returns
+    ///
+    /// * [AssetId] - The AssetId of the asset. Computed by hashing the ContractId and SubId.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// use std::{callframes::contract_id, constants::ZERO_B256};
+    ///
+    /// fn foo() {
+    ///     let contract_id = contract_id();
+    ///     let sub_id = ZERO_B256;
+    ///
+    ///     let asset_id = AssetId::new(contract_id, sub_id);        
+    /// }
+    /// ```
+    pub fn new(contract_id: ContractId, sub_id: SubId) -> Self {
+        let value = sha256((contract_id, sub_id));
+        Self { value }
+    }
+
+    /// Creates a new AssetId from a ContractId and the zero SubId.
+    ///
+    /// # Arguments
+    ///
+    /// * `contract_id`: [ContractId] - The ContractId of the contract that created the asset.
+    ///
+    /// # Returns
+    ///
+    /// * [AssetId] - The AssetId of the asset. Computed by hashing the ContractId and the zero SubId.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// use std::{callframes::contract_id, constants::ZERO_B256};
+    ///
+    /// fn foo() {
+    ///     let contract_id = contract_id();
+    ///     let sub_id = ZERO_B256;
+    ///
+    ///     let asset_id = AssetId::standard(contract_id);
+    ///
+    ///     assert(asset_id == AssetId::new(contract_id, sub_id));
+    /// }
+    /// ```
+    pub fn standard(contract_id: ContractId) -> Self {
+        let value = sha256((contract_id, ZERO_B256));
+        Self { value }
+    }
+
+    /// Represents bridged Ether on the main Fuel Network. Can be configured to represent another asset on another instance of the Fuel network.
+    ///
+    /// # Additional Information
+    ///
+    /// The base asset id is minted from the zero ContractId with the zero SubId.
+    /// It is computed as sha256((ZERO_B256, ZERO_B256))).
+    ///
+    /// NOT equal to ZERO_B256.
+    ///
+    /// # Returns
+    ///
+    /// * [AssetId] - The AssetId of the asset. Computed by hashing the zero ContractId and the zero SubId.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// use std::token::transfer;
+    ///
+    /// fn foo() {
+    ///     let asset_id = AssetId::base_asset_id();
+    ///     let amount = 100;
+    ///     let recipient = Identity::ContractId(ContractId::from(ZERO_B256));
+    ///
+    ///     transfer(recipient, asset_id, amount);
+    /// ```
+    pub fn base_asset_id() -> Self {
+        Self {
+            // sha256((ZERO_B256, ZERO_B256))). Prehashed to save gas.
+            value: 0xf5a5fd42d16a20302798ef6ed309979b43003d2320d9f0e8ea9831a92759fb4b,
+        }
+    }
 }
+
+/// Represents bridged Ether on the main Fuel Network. Can be configured to represent another asset on another instance of the Fuel network.
+pub const BASE_ASSET_ID: AssetId = AssetId {
+    // sha256((ZERO_B256, ZERO_B256))). Prehashed to save gas.
+    value: 0xf5a5fd42d16a20302798ef6ed309979b43003d2320d9f0e8ea9831a92759fb4b,
```

Comment:
For the base asset, it doesn't rely on any contract id and currently the vm still hardcodes it as a `ZERO_B256`